### PR TITLE
enable centos9 builds for ceph-iscsi, tcmu-runner

### PR DESCRIPTION
**Pull request by cbodley**
adds centos9 to AVAILABLE_DIST for these packages

the build script for nfs-ganesha looked like the only one that needed a small change to handle centos9

### CODE REVIEW

**Comment by cbodley** [comment]
@dmick @zmc how can we test these builds?

**Comment by dmick** [comment]
that'd be a question for the code owners.  I don't know who those are.

**Comment by cbodley** [comment]
sorry, not test the packages, but how to trigger the builds to see if my script changes work? i'm willing to iterate on this but don't know how to schedule builds against a custom ceph-build branch

**Comment by dmick** [comment]
Ah, I get you.  One can upload a test version of the job (which I usually do by naming it "preserve-dmick-<job>" to keep the ceph-build job from deleting it) by running jenkins-jobs by hand.  This is not for the faint of heart, however, and requires jenkins secrets to run, so it's probably best left to me.  I can create the jobs for you and you can experiment with them.  I'll DM.

**Comment by cbodley** [comment]
initial test results:
* https://jenkins.ceph.com/job/preserve-cbodley-ceph-iscsi/ was successful

* https://jenkins.ceph.com/job/preserve-cbodley-tcmu-runner failed on both [x86_64](https://jenkins.ceph.com/job/preserve-cbodley-tcmu-runner/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos9,DIST=centos9,MACHINE_SIZE=huge/lastBuild/consoleFull) and [arm64](https://jenkins.ceph.com/job/preserve-cbodley-tcmu-runner/ARCH=arm64,AVAILABLE_ARCH=arm64,AVAILABLE_DIST=centos9,DIST=centos9,MACHINE_SIZE=huge/lastBuild/consoleFull) builds with:
> No matching package to install: 'glusterfs-api-devel'

* https://jenkins.ceph.com/job/preserve-cbodley-nfs-ganesha/ failed with cmake errors:
> CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
NSL_LIBRARY (ADVANCED)
    linked by target "ntirpc" in directory /home/jenkins-build/build/workspace/preserve-cbodley-nfs-ganesha/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos9/DIST/centos9/MACHINE_SIZE/huge/nfs-ganesha/src/libntirpc/src

**Comment by dmick** [comment]
I've confirmed that glusterfs-api-devel is not shipped in centos 9.  We can petition for it to be included, or we can (possibly) dig it out of a koji ... thing, or we can compile tcmu-runner without gluster support.  (The latter seems easiest and probably sufficient.)

**Comment by dmick** [comment]
FWIW, https://jenkins.ceph.com/job/preserve-cbodley-tcmu-runner/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos9,DIST=centos9,MACHINE_SIZE=huge/3/console is a build without glfs support; appears to have worked

**Comment by cbodley** [comment]
thanks @dmick, did that involve further changes to this branch?

@idryomov @lxbsz what do you guys think? is that glusterfs support important for our centos9 tcmu-runner packages?

**Comment by idryomov** [comment]
> is that glusterfs support important for our centos9 tcmu-runner packages?

It's not needed at all.  All we care about here is RBD.

**Comment by dmick** [comment]
@cbodley yes:  

```
diff -r tcmu-runner/build/build_rpm preserve-cbodley-tcmu-runner/build/build_rpm
90,91c90,91
< sudo mock --verbose -r tcmu-runner.cfg --scrub=all
< sudo mock --verbose -r tcmu-runner.cfg --resultdir=$WORKSPACE/dist/RPMS/ ${SRPM} || ( tail -n +1 $WORKSPACE/dist/RPMS/{root,build}.log && exit 1 )
---
> sudo mock --verbose --without glfs -r tcmu-runner.cfg --scrub=all
> sudo mock --verbose --without glfs -r tcmu-runner.cfg --resultdir=$WORKSPACE/dist/RPMS/ ${SRPM} || ( tail -n +1 $WORKSPACE/dist/RPMS/{root,build}.log && exit 1 )

```

**Comment by dmick** [comment]
What's the plan for ganesha?

File: tcmu-runner/build/build_rpm [hunk 1]
```diff
@@ -86,9 +86,10 @@ echo "\"\"\"" >> tcmu-runner.cfg
 cat tcmu-runner.cfg
 
 ## Build the binaries with mock
+## disable 'glfs' because centos9 packages aren't available
 echo "Building RPMs"
-sudo mock --verbose -r tcmu-runner.cfg --scrub=all
-sudo mock --verbose -r tcmu-runner.cfg --resultdir=$WORKSPACE/dist/RPMS/ ${SRPM} || ( tail -n +1 $WORKSPACE/dist/RPMS/{root,build}.log && exit 1 )
+sudo mock --verbose --without glfs -r tcmu-runner.cfg --scrub=all
+sudo mock --verbose --without glfs -r tcmu-runner.cfg --resultdir=$WORKSPACE/dist/RPMS/ ${SRPM} || ( tail -n +1 $WORKSPACE/dist/RPMS/{root,build}.log && exit 1 )
```
Comment (lxbsz) [inline tcmu-runner/build/build_rpm:92]: Maybe you should also disable `qcow,file,zbc` :

```
o- backstores .......................................................... [...]
  o- user:glfs .......................................... [Storage Objects: 0]
  o- user:qcow .......................................... [Storage Objects: 0]
  o- user:rbd ........................................... [Storage Objects: 0]
  o- user:file .......................................... [Storage Objects: 0]
  o- user:zbc ........................................... [Storage Objects: 0]
```

Comment (dmick) [reply tcmu-runner/build/build_rpm:92]: Maybe, but, those still build, and might be useful for some user(s).

**Comment by cbodley** [comment]
> What's the plan for ganesha?

@dmick in [#sepia slack](https://ceph-storage.slack.com/archives/C1HFJ4VTN/p1701439349395339), @kalebskeithley suggested that we use the ganesha packages from the centos storage sig:

> After much go-around I think I finally convinced Blaine Gardner to use the ganesha rpms from the CentOS Storage SIG for their CI.  I don't really want to get in a position where we have yet another set of ganesha builds that are going to get out of date and someone will have to remember to build them after a ganesha release. I'd really like to see ceph/teuthology do the same thing. They'll always be up to date with the latest released version. The only thing you'll need to change is from ganesha-5 to ganesha-6 (and ganesha-7) at some point.

the [Containerfile](https://github.com/ceph/ceph/pull/54575/files#diff-4b612b8de8bfe0eec09a16cd8002b7a0cc8908ba5672931e083b71674458ff18R44-R50) that @BlaineEXE wrote in https://github.com/ceph/ceph/pull/54575 already uses that for repo centos8. eventually we probably want to converge on using that Containerfile for upstream container builds (and update that to use a centos9 base). in the meantime, i guess we'd need to update the ceph-container scripts to use that storage sig repo, and eventually drop the ganesha builds from ceph-build?

if there are other obstacles to doing that, i could ping Kaleb/Frank to help work through the ganesha build issues above

**Comment by dmick** [comment]
I have begged for literally years for someone to take responsibility for "which version of ganesha to include", and so if that's an implementable strategy, it's more than fine by me.

File: tcmu-runner/build/build_rpm (hunk 1 above)
Comment (cbodley) [reply tcmu-runner/build/build_rpm:92]: i had pushed a commit to disable other backends too, but hadn't seen @dmick's comment; i agree and reverted back to the version that Dan already tested

**Comment by dmick** [comment]
so....do we wanna drop the ganesha changes from this PR in light of the above?

**Comment by cbodley** [comment]
i dropped the ganesha commit

**Review by idryomov — APPROVED** [review]

**Review by dmick — APPROVED** [review]

**Comment by cbodley** [comment]
@dmick does this summarize the next steps?

* change ceph-container to consume ganesha packages from the centos storage sig repo
* add ceph-container step to centos9 builds

anything else? anything you need from me?

**Comment by dmick** [comment]
That's all I can think of.  Let me dig around and see what code I had lying about in half-tested form

**Pull request merged**